reflex/app.py

```diff
@@ -1002,7 +1002,8 @@ async def upload_file(request: Request, files: List[UploadFile]):
             )
 
         # Get the state for the session.
-        state = await app.state_manager.get_state(token)
+        substate_token = token + "_" + handler.rpartition(".")[0]
```

Comment:
nit: maybe we can pull this out into a function `get_substater_token(token, handler)`